src/main/java/com/openshift/internal/restclient/authorization/AuthorizationContext.java

```diff
@@ -35,12 +46,34 @@ public AuthorizationContext(String token, String expires, IUser user, String sch
 		this.scheme = scheme;
 	}
 	
+	public AuthorizationContext clone() {
+		AuthorizationContext context = new AuthorizationContext(this.token, this.expires, this.user, this.scheme);
+		context.setUserName(this.userName);
+		context.setPassword(this.password);
+		context.setClient(this.client);
+		return context;
+	}
+	
+	public void setClient(IClient client){
+		this.client = client;
+	}
+	
 	@Override
 	public boolean isAuthorized() {
+		if(user == null) {
+			synchronized (this) {
+				user = client.get(ResourceKind.USER, "~", "");
```

Comment:
That is not strictly true as before we assumed you were authorized if your token was not null which is not necessarily correct; it just means you have a token.  In JBT we had logic that more or less did what was doing here so you were making a remote call anyway.  Since we simplified how the auth logic is consumed, I believe this is a reasonable trade off. At some point or another, remote calls are going to need to be made in the authorization process.